test/integ/routers/alpha-router/alpha-router.integration.test.ts

```diff
@@ -0,0 +1,995 @@
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quote,
+            methodParameters,
+            route,
+          } = convertSwapDataToResponse(amount, tradeType, swap!)
+
+          expect(methodParameters).not.toBeUndefined;
+          expect(route).not.toBeUndefined
+
+          const amountInEdgesTotal = _(route)
+            .flatMap((route) => route[0]!)
+            .filter((pool) => !!pool.amountIn)
+            .map((pool) => BigNumber.from(pool.amountIn))
+            .reduce((cur, total) => total.add(cur), BigNumber.from(0))
+          const amountIn = BigNumber.from(quote)
+          expect(amountIn.eq(amountInEdgesTotal))
+
+          const amountOutEdgesTotal = _(route)
+            .flatMap((route) => route[0]!)
+            .filter((pool) => !!pool.amountOut)
+            .map((pool) => BigNumber.from(pool.amountOut))
+            .reduce((cur, total) => total.add(cur), BigNumber.from(0))
+          const amountOut = BigNumber.from(quote)
+          expect(amountOut.eq(amountOutEdgesTotal))
+
+          const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } = await executeSwap(
+            methodParameters!,
+            tokenIn,
+            tokenOut!
+          )
+
+          if (tradeType == TradeType.EXACT_INPUT) {
+            expect(tokenInBefore.subtract(tokenInAfter).toExact()).toEqual('1000000')
+            checkQuoteToken(tokenOutBefore, tokenOutAfter, CurrencyAmount.fromRawAmount(tokenOut, quote))
+          } else {
+            // Hard to test ETH balance due to gas costs for approval and swap. Just check tokenIn changes
+            checkQuoteToken(tokenInBefore, tokenInAfter, CurrencyAmount.fromRawAmount(tokenIn, quote))
+          }
+        })
+
+        /**
+         * Skipped @see https://github.com/Uniswap/smart-order-router/issues/94
+         */
+        xit(`eth -> erc20`, async () => {
+          const tokenIn = Ether.onChain(1) as Currency;
+          const tokenOut = UNI_MAINNET;
+          const amount = tradeType == TradeType.EXACT_INPUT ?
+            parseAmount('10', tokenIn)
+            : parseAmount('10000', tokenOut);
+
+          const swap = await alphaRouter.route(
+            amount,
+            getQuoteToken(tokenIn, tokenOut, tradeType),
+            tradeType,
+            {
+              recipient: alice._address,
+              slippageTolerance: SLIPPAGE,
+              deadline: parseDeadline(360),
+            },
+            {
+              ...ROUTING_CONFIG,
+              protocols: [Protocol.V2]
+            }
+          );
+          expect(swap).toBeDefined();
+          expect(swap).not.toBeNull();
+
+          const {
+            quote,
+            methodParameters
+          } = convertSwapDataToResponse(amount, tradeType, swap!)
+
+          expect(methodParameters).not.toBeUndefined;
+
+          const { tokenInBefore, tokenInAfter, tokenOutBefore, tokenOutAfter } = await executeSwap(
```

Comment:
`validateStandardExecuteSwap` (ik this test isnt working but will be nice to have every test basically follow the same format